src/state/variantEditor.slice.ts

```diff
@@ -0,0 +1,87 @@
+import { type RouterOutputs } from "~/utils/api";
+import { type SliceCreator } from "./store";
+import loader from "@monaco-editor/loader";
+import openAITypes from "~/codegen/openai.types.ts.txt";
+
+export type VariantEditorSlice = {
```

Comment:
Nit: I would maybe name this something like "VariantEditorModelSlice", since it doesn't contain information that we would store in the variant editor itself, such as the prompt-construction function.

Reply:
renamed to `sharedVariantEditor` to hopefully make it more clear this is general init code not a specific editor instantiation!